### PR TITLE
Add `required` job in kvrocks CI for `required_status_checks` configuration

### PR DESCRIPTION
**Pull request by PragmaTwice**
refer to https://github.com/apache/incubator-kvrocks/pull/635#issuecomment-1158449828

### CODE REVIEW

File: .asf.yaml [hunk 1]
```diff
@@ -31,6 +31,10 @@ github:
       required_pull_request_reviews:
         dismiss_stale_reviews: false
         required_approving_review_count: 1
+      required_status_checks:
+        strict: true
+        contexts:
+          - "kvrocks ci actions / required (pull_request)"
```
Comment (tisonkun) [inline .asf.yaml]: I think `pull_request` is unneeded. You can refer to https://github.com/apache/skywalking/blob/master/.asf.yaml.

File: .github/workflows/kvrocks.yaml [hunk 2]
```diff
@@ -96,3 +96,9 @@ jobs:
           cd build
           cmake ..
           make -j4
+
+  required:
+    runs-on: ubuntu-latest
+    needs: [lint-build-test-on-ubuntu, build-on-macos-latest]
+    steps:
+      - run: echo success
```
Comment (tisonkun) [inline .github/workflows/kvrocks.yaml]: You may have to check needs.dep in this step. Also you can refer to skywalking's ci yaml file.

File: .asf.yaml (hunk 1 above)
Comment (PragmaTwice) [reply .asf.yaml]: done

File: .github/workflows/kvrocks.yaml (hunk 2 above)
Comment (PragmaTwice) [reply .github/workflows/kvrocks.yaml]: done

**Review by tisonkun — APPROVED** [review]
LGTM. Merging...

**Pull request merged**